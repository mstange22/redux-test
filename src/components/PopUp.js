import React from 'react';
import PropTypes from 'prop-types';

const PopUp = ({ title, message }) => (
  <div
    style={{
      textAlign: 'center',
      height: 100,
      width: 400,
      position:
      'fixed',
      top: 200,
      left: '50%',
      marginLeft: -220,
      backgroundColor: '#f4f442',
      border: 'solid 1px black',
      borderRadius: 10,
      display: 'flex',
      justifyContent: 'flex-start',
      padding: 20,
      flexDirection: 'column',
    }}
  >
    <div className="popup-title" style={{ fontWeight: 700 }}>
      {`Pop-Up Says "${title}"`}
    </div>
    <br />
    <div className="popup-message">
      {message}
    </div>
  </div>
);

PopUp.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
};

export default PopUp;
