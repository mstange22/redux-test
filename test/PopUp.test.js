import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PopUp from '../src/components/PopUp';

configure({ adapter: new Adapter() });

const props = { title: 'Pop-Up Title', message: 'Your message goes here' };
const shallowComponent = shallow(<PopUp {...props} />);

describe('<Pop-Up />', () => {
  test('snapshot test', () => {
    const component = renderer.create(
      <PopUp {...props} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  test('exists', () => {
    expect(shallowComponent.exists()).toBe(true);
  });
  test('title', () => {
    expect(shallowComponent.find('.popup-title').text()).toBe(`Pop-Up Says "${props.title}"`);
  });
  test('message', () => {
    expect(shallowComponent.find('.popup-message').text()).toBe(props.message);
  });
});
