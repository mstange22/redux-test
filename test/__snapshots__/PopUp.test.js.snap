// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pop-Up /> snapshot test 1`] = `
<div
  className="pop-up"
  style={
    Object {
      "backgroundColor": "#f4f442",
      "border": "solid 1px black",
      "borderRadius": 10,
      "display": "flex",
      "flexDirection": "column",
      "height": 100,
      "justifyContent": "flex-start",
      "left": "50%",
      "marginLeft": -220,
      "padding": 20,
      "position": "fixed",
      "textAlign": "center",
      "top": 200,
      "width": 400,
    }
  }
>
  <div
    className="popup-title"
    style={
      Object {
        "fontWeight": 700,
      }
    }
  >
    Pop-Up Says "Pop-Up Title"
  </div>
  <br />
  <div
    className="popup-message"
  >
    Your message goes here
  </div>
</div>
`;
